src/applications/hca/components/IDForm.jsx

```diff
@@ -75,6 +75,47 @@ export default class IDForm extends React.Component {
     },
   };
 
+  renderServerErrorAlert = () => {
+    const { errors } = this.props;
+    if (errors && errors.some(error => error.code === '500')) {
```

Comment:
I'm gonna take a stab at refactoring this component with more descriptive prop names, with the goal of avoiding doing this kind of logic in the component code. Would be simpler if the `render` method could simply do something like `{this.props.showServerError && this.renderServerError()}`.

Reply:
sounds good- I'm confused by the component composition between IDPage, IDForm, and SchemaForm. IDForm has some state but doesn't have a `constructor`- not clear what its role is. 

I would try to compose IDForm and SchemaForm in IDPage instead of drilling props through IDForm. Might be able to move the schema stuff from IDForm and put into consts. 

Might be able to just get rid of IDForm, create props like `showServerError` in the IDPage's `mapStateToProps` and use `SchemaForm` directly in `IDPage`. I don't see a big advantage to spreading this logic across two small components like this.

Reply:
The constructor really isn't needed with React components (https://hackernoon.com/the-constructor-is-dead-long-live-the-constructor-c10871bea599) so I didn't put one in the IDForm. 

Fair point on the prop drilling. I don't remember if I set it up this way because I thought I was going to have to reuse the IDForm elsewhere or if it was just my instinct to break the two apart. I'll see how it looks to just merge the two.